aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1126,6 +1126,25 @@ Tensor trace_cpu(const Tensor& self) {
   return result;
 }
 
+// TODO: this routine should be implemented without diag and sum for perf problems,
+// see https://github.com/pytorch/pytorch/pull/47305,
+Tensor linalg_trace(const Tensor& self, int64_t offset) {
+  TORCH_CHECK(self.dim() >= 2,
+           "self should have at least 2 dimensions, but has ", self.dim(), " dimensions instead");
```

Comment:
The user documented name is `input` (per your docs below) and these warnings should start with the name of the operation like this:

```
torch.linalg.trace(): input should have at least...
```

Reply:
It might be nice to change the user-facing name of this argument to `A`, which is the name we use throughout `torch.linalg`